core/MyGatewayTransportEthernet.cpp

```diff
@@ -49,7 +49,9 @@ MyMessage _ethernetMsg;
 #define ARRAY_SIZE(x)  (sizeof(x)/sizeof(x[0]))
 
 typedef struct {
+	// cppcheck-suppress unusedStructMember
```

Comment:
Should we (manually, during code review I guess) encourage that the annotations are accompanied by a comment explaining why the suppression was added?

To take this code line as example, why do we want to suppress the warning?

If we decide to encourage commenting, should the comment be before or after the annotation? On the same line or above/below? /* style comment or // ? Consistency would make it easier for code contributors and reviewers.

Reply:
I can add further comments on the inline suppressions. Most of these are due to them being initialized in a init function, and to avoid claiming unnecessary memory because the class is instantiated but not used.

Reply:
In this case, I think it is a matter of the inputBuffer is used in a rats-nest of #ifdef:s which falls between the masks of the cppcheck grid (it will not evaluate ALL possible combinations of preprocessor flags).